### PR TITLE
[Yamato] Pipeline reorganization

### PR DESCRIPTION
**Pull request by kivistikliis**
### Purpose of this PR
Reorganizes Yamato test architecture according to https://jira.unity3d.com/browse/SDET-189 
Job hierarchy per area from highest to lowest goes: `Nightly `> `<area or project>` > (`PR` or `Extra`)> individual jobs. 
`Nighty` jobs exist only for ShaderGraph (`shadergraph.metafile`), URP (new `urp.metafile`), HDRP (`hdrp.metafile`), and VFX (new `vfx.metafile`) areas, and each of these nightlies encompasses the rest of their dependencies.

`Nightly` contains the nightlies of the 4 above-mentioned areas
`ABV (all PR jobs)` contains the PR jobs of the 4 above-mentioned areas

---
### Testing status
Checked that all dependencies are correct, and the 4 area entrypoints for CI are according to the proposed architecture
- [Nightly HDRP](https://yamato.cds.internal.unity3d.com/jobs/902-Graphics/tree/yamato%252Fpipeline-reorg/.yamato%252Fall-hdrp.yml%2523Nightly_HDRP_trunk/7045762/job/pipeline)
- [Nightly ShaderGraph](https://yamato.cds.internal.unity3d.com/jobs/902-Graphics/tree/yamato%252Fpipeline-reorg/.yamato%252Fall-shadergraph.yml%2523Nightly_ShaderGraph_trunk/7045788/job/pipeline)
- [Nightly URP](https://yamato.cds.internal.unity3d.com/jobs/902-Graphics/tree/yamato%252Fpipeline-reorg/.yamato%252Fall-urp.yml%2523Nightly_URP_trunk/7046027/job/pipeline)
- [Nightly VFX](https://yamato.cds.internal.unity3d.com/jobs/902-Graphics/tree/yamato%252Fpipeline-reorg/.yamato%252Fall-vfx.yml%2523Nightly_VFX_trunk/7046108/job/pipeline)
- [General checks](https://yamato.cds.internal.unity3d.com/jobs/902-Graphics/tree/yamato%252Fpipeline-reorg/.yamato%252F_abv.yml%2523General_Checks_trunk/7078755/job) (formatting was added in a later commit than the above jobs)

---
### Comments to reviewers
- Triggers exist for:
  - **Nightly**
  -  **ABV (all PR jobs)** 
  - **General checks**
  - **PR HDRP**
  - **PR URP**
  - **PR VFX**
  - **PR SG**
  - PR PostProcessing 
  - PR SmokeTest
  - PR urp-upgrade (trigger is left how it was, the project/package will soon be included into the main CI pipelines)
- The split between PR/Extra is right now based on the existing PR/Nightly split, thus not all pipelines are included in the automated CI
- Requires https://github.cds.internal.unity3d.com/unity/gfx-sdet-tools/pull/43 
- Some `Extra` jobs are empty for now, but kept for quicker setup in future
- Perf jobs have a single entry point (no separation between PR/Extra) and are not included in the 4 PR entrypoints

### CODE REVIEW

File: .yamato/config/abv.metafile [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  name: ABV
+
+jobs:
+  - name: Nightly
+    dependencies:
+      - .yamato/all-hdrp.yml#Nightly_HDRP_<TRACK>
+      - .yamato/all-vfx.yml#Nightly_VFX_<TRACK>
+      - .yamato/all-urp.yml#Nightly_URP_<TRACK>
+      - .yamato/all-shadergraph.yml#Nightly_ShaderGraph_<TRACK>
+      - <PROJECT_ALL_YML_FILE>#General_Checks_<TRACK>
```
Comment (theo-pnv) [inline .yamato/config/abv.metafile:11]: What does PROJECT_ALL_YML_FILE refer to? (where is it set?)

**Review by theo-pnv — COMMENTED** [review]
Awesome work, thanks a lot!
Nitpicky thought: do you think it would be worth trying to have the same name for all the urp pipelines? I feel like it would be more consistent than to have URP sometimes and Universal other times. Drawback: yet another change of names/and breaks the yamato history.
image

File: .yamato/_abv.yml [hunk 2]
```diff
@@ -8,56 +8,47 @@
...
+      -  path: .yamato/all-urp.yml#Nightly_URP_trunk
          rerun: on_new_revision
-      -  path: .yamato/all-urp_performance_lostcrypt.yml#PR_URP_Performance_LostCrypt_trunk
+      -  path: .yamato/all-shadergraph.yml#Nightly_ShaderGraph_trunk
          rerun: on_new_revision
-      -  path: .yamato/all-hdrp_performancetests.yml#PR_HDRP_PerformanceTests_trunk
+      -  path: .yamato/_abv.yml#General_Checks_trunk
          rerun: on_new_revision
-      -  path: .yamato/all-vikingvillage_builtin.yml#PR_VikingVillage_BuiltIn_trunk
+General_Checks_trunk:
```
[the remark this answers is not in the document]
Comment (kivistikliis) [reply .yamato/_abv.yml:33]: Good catch, I believe so. I will add it, thanks!

File: .yamato/all-urp_lighting.yml [hunk 3]
```diff
@@ -30,16 +32,12 @@ Nightly_URP_Lighting_trunk:
          rerun: on_new_revision
       -  path: .yamato/urp_lighting-linux-vulkan.yml#URP_Lighting_Linux_Vulkan_playmode_cache_mono_Linear_trunk
          rerun: on_new_revision
-      -  path: .yamato/urp_lighting-win-dx11.yml#URP_Lighting_Win_DX11_Standalone_XR_cache_mono_Linear_trunk
```
[the remark this answers is not in the document]
Comment (kivistikliis) [reply .yamato/all-urp_lighting.yml:33]: Another good catch, will add them back in and doublecheck the rest of the split projects as well.

File: .yamato/config/abv.metafile (hunk 1 above)
Comment (kivistikliis) [reply .yamato/config/abv.metafile:11]: It refers to `.yamato/all-{project}.yml` for the corresponding metafile (the abv metafile works similar to projects now, but is a specific case and outputs to `.yamato/_abv.yml` still), that comes with https://github.cds.internal.unity3d.com/unity/gfx-sdet-tools/pull/43 . I can see the confusion though, I'm fine with changing this to the concrete path directly

Comment (theo-pnv) [reply .yamato/config/abv.metafile:11]: I see, that's fine don't need to worry about updating both PRs

**Review by theo-pnv — APPROVED** [review]

**Review by sophiaaar — APPROVED** [review]

**Pull request merged**